__tests__/services/VoiceService.test.ts

```diff
@@ -0,0 +1,219 @@
+import React from "react";
+import { expect } from "@jest/globals";
+import {
+  startVoiceRecording,
+  stopVoiceRecording,
+  syncVoiceWithChatInput,
+} from "../../src/services/VoiceService";
+import { Settings } from "../../src/types/Settings";
+import { RefObject } from "react";
+
+describe("VoiceService", () => {
+  let mockInputRef: RefObject<HTMLInputElement | null>;
+
+  beforeAll(() => {
+    // Mock MediaStream
+    class MockMediaStream {
+      active = true;
+      id = "mock-stream";
+      getTracks() {
+        return [];
+      }
+      addTrack() {}
+      removeTrack() {}
+    }
+    global.MediaStream = MockMediaStream as unknown as typeof MediaStream;
+
+    // Mock MediaRecorder
+    class MockMediaRecorder {
+      state = "inactive";
+      start = jest.fn();
+      stop = jest.fn();
+      ondataavailable: ((event: any) => void) | null = null;
+      onstop: (() => void) | null = null;
+
+      constructor() {}
+    }
+    global.MediaRecorder = MockMediaRecorder as unknown as typeof MediaRecorder;
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    mockInputRef = { current: document.createElement("input") };
+
+    // Mock SpeechRecognition
+    Object.defineProperty(window, "SpeechRecognition", {
+      configurable: true,
+      value: jest.fn().mockImplementation(() => ({
+        start: jest.fn(),
+        stop: jest.fn(),
+        onresult: jest.fn(),
+        onend: jest.fn(),
+      })),
+    });
+
+    // Mock navigator.mediaDevices
+    Object.defineProperty(navigator, "mediaDevices", {
+      configurable: true,
+      value: {
+        getUserMedia: jest.fn(),
+      },
+    });
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe("SpeechRecognition", () => {
+    it("starts voice recording with SpeechRecognition", () => {
+      const mockToggleVoice = jest.fn();
+      const mockTriggerSendVoiceInput = jest.fn();
+      const mockSetTextAreaValue = jest.fn();
+      const mockSetInputLength = jest.fn();
+      const mockAudioChunksRef: RefObject<BlobPart[]> = { current: [] };
+
+      const mockSettings: Settings = {
+        voice: {
+          sendAsAudio: false,
+          language: "en-US",
+          timeoutPeriod: 5000,
+          autoSendPeriod: 3000,
+          autoSendDisabled: false,
+        },
+      };
+
+      startVoiceRecording(
+        mockSettings,
+        mockToggleVoice,
+        mockTriggerSendVoiceInput,
+        mockSetTextAreaValue,
+        mockSetInputLength,
+        mockAudioChunksRef,
+        mockInputRef
+      );
+
+      expect(mockToggleVoice).not.toHaveBeenCalled();
```

Comment:
This is not sufficient to verify that the function works as intended, it'll be great to check for `voiceToggledOn` too.